amida-tech/indaba-client
# Bug/inba 506 taskoptions reassign

#### What's this PR do?
Makes the "reassign task" option in the Task Options modal actually work as intended.

#### Related JIRA tickets:
Sits atop of INBA-505, and will be related to INBA-507 and INBA-508 (force completing tasks and notifying a user respectively).

#### How should this be manually tested?
Login as a test admin. If you haven't already, create a project that has some tasks assigned.

Go into Project Manager. Click on the task options modal button (...). When displayed, click on the "Reassign" option.

In the dropdown which displays users, ensure that the users:
1) Are members of the user group that the task's stage permits. 
2) That the current assigned user has the label "(Currently Assigned)" next to it.

Select a user who isn't assigned. **Unclick the "notify user" checkbox.** Click save.

Ensure that the backend updates with the user's id (in the `userIds`) and that the table displays the change in the name. 

#### Any background context you want to provide?
Slightly reduced some of the unnecessary complexities of task reducer. But yeah, I agree with Chris that we should eventually refactor that into the `projectReducer`.

#### Screenshots (if appropriate):
